File: Tree/tree.js
var Tree = function(value) {
  this.value = value;
  this.children = [];
};

Tree.prototype.addChild = function(value) {
  this.children.push(new Tree(value));
};

Tree.prototype.contains = function(value) {
  if (this.value === value) {
    return true;
  } else if (this.children.length) {
    var child = this.children;
    for (var i = 0, l = child.length; i < l; i++) {
      return child[i].contains(value) && true;
    }
  }
  return false;
};

File: Tree/tree.rb
class Tree
  attr_reader :value, :children

  def initialize(value)
    @value = value
    @children = []
  end

  def add_child(value)
    @children.push(Tree.new(value))
  end
  def contains(value)
    if @value == value
      return true
    elsif (@children.length > 0)
      @children.each do |child|
        return true if child.contains(value)
      end
    end
    return false
  end
end
